Added nether biome whitelist/blacklist config option

caelwarner opened the pull request:
- Allows players to create a whitelist or blacklist of biomes that spawn in the nether
- Added WorldGenerationScreen GUI

caelwarner committed:
Added nether biome whitelist/blacklist config option

- Allows players to create a whitelist or blacklist of biomes that spawn in the nether
- Added WorldGenerationScreen GUI

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/infernalexp/Infernal-Expansion?pullRequest=155) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/infernalexp/Infernal-Expansion?pullRequest=155) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/infernalexp/Infernal-Expansion?pullRequest=155) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/infernalexp/Infernal-Expansion?pullRequest=155) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/infernalexp/Infernal-Expansion?pullRequest=155) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/infernalexp/Infernal-Expansion?pullRequest=155) it.</sub>

Dariensg commented:
LGTM, we need to make sure that all the other lang files are also updated to reflect this. I can take care of that in a matter of minutes once this is merged if you would like.

caelwarner commented:
I say lets wait until we have actually finalized everything and are ready to release the beta.

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/world/dimension/ModNetherBiomeCollector.java:
```diff
@@ -16,13 +18,17 @@
             if (entry.getValue().getCategory() == Biome.Category.NETHER && !entry.getKey().getLocation().getNamespace().equals("ultra_amplified_dimension")) {
                 if (!netherBiomeList.contains(entry.getKey())) {
-                    netherBiomeList.add(entry.getKey());
+                    if ((whitelistEnabled && whitelistOrBlacklist.contains(entry.getKey().getLocation().toString())) || (!whitelistEnabled && !whitelistOrBlacklist.contains(entry.getKey().getLocation().toString())) || whitelistOrBlacklist.isEmpty()) {
```
would've preferred some line breaks/indentation here

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java:
```diff
@@ -282,17 +286,32 @@
+
+        biomesWhitelistEnabled = builder
+            .comment("Should the biome list below act as a whitelist (True/On), or a blacklist (False/Off). CHANGING THIS REQUIRES A GAME RESTART.")
+            .translation(InfernalExpansion.MOD_ID + ".config.tooltip.biomesWhitelistEnabled")
```
```suggestion
            .comment("Determines if the biome list below should act as a whitelist, or, when disabled, a blacklist.")
            .comment("CHANGING THIS REQUIRES A GAME RESTART.")
            .translation(InfernalExpansion.MOD_ID + ".config.tooltip.biomesWhitelistEnabled")
```

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java:
```diff
@@ -50,7 +50,11 @@
-    
+
+    // World Generation
+    final ForgeConfigSpec.BooleanValue biomesWhitelistEnabled;
```
`biomeListIsWhitelist` would be more appropriate and self-explanatory.

SwanX1 commented on src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java:
```diff
@@ -50,7 +50,11 @@
+
+    // World Generation
+    final ForgeConfigSpec.BooleanValue biomesWhitelistEnabled;
+    final ForgeConfigSpec.ConfigValue<String> biomesWhitelistOrBlacklist;
```
in `ForgeConfigSpec.ConfigValue<T>`, `T` can be `List<String>`

SwanX1 replied:
If `biomesWhitelistEnabled` is renamed, rename this to `biomesList`

caelwarner replied:
Does making it `List<String>` parse it automatically?

SwanX1 replied:
yes

caelwarner replied:
honestly, keeping it as a string is just easier for me. If I change it to be a list then I have to write code to parse it for the GUI

SwanX1 reviewed (changes requested):
minor changes

caelwarner committed:
Minor fixes

- Refactored biomesWhitelistEnabled to biomesListIsWhitelist
- Refactored biomesWhitelistOrBlacklist to biomesList
- Small spacing changes

SwanX1 commented:
Gonna merge this one
